File: src/main/java/cn/edu/zjnu/acm/judge/Application.java
/*
 * Copyright 2015 zhanhb.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package cn.edu.zjnu.acm.judge;

import com.google.common.base.Strings;
import java.util.Locale;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.Banner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.builder.SpringApplicationBuilder;
import org.springframework.boot.context.embedded.EmbeddedWebApplicationContext;
import org.springframework.boot.context.web.SpringBootServletInitializer;
import org.springframework.context.ApplicationContext;

@Slf4j
@SpringBootApplication
public class Application extends SpringBootServletInitializer {

    public static void main(String[] args) {
        ApplicationContext context = SpringApplication.run(Application.class, args);
        if (log.isInfoEnabled() && context instanceof EmbeddedWebApplicationContext) {
            int port = ((EmbeddedWebApplicationContext) context).getEmbeddedServletContainer().getPort();
            String contextPath = context.getApplicationName();
            String url = String.format(Locale.US, "http://localhost:%d%s", port, contextPath);
            String dashes = Strings.repeat("-", 72);
            log.info("Access URLs:\n{}\n\tLocal:\t\t{}\n{}", dashes, url, dashes);
        }
    }

    @Override
    protected SpringApplicationBuilder configure(SpringApplicationBuilder builder) {
        return builder.sources(Application.class).bannerMode(Banner.Mode.OFF);
    }

}

File: site/src/main/kotlin/cn/edu/zjnu/acm/judge/Application.kt
package cn.edu.zjnu.acm.judge

import com.google.common.base.Strings
import org.slf4j.LoggerFactory
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.boot.web.context.WebServerApplicationContext
import org.springframework.cache.annotation.EnableCaching

/**
 * @author zhanhb
 */
@EnableCaching
@SpringBootApplication
class Application

private val log = LoggerFactory.getLogger(Application::javaClass.javaClass)

fun main(args: Array<String>) {
    val context = runApplication<Application>(*args)
    if (log.isInfoEnabled && context is WebServerApplicationContext) {
        val port = context.webServer.port
        val contextPath = context.getApplicationName()
        val dashes = Strings.repeat("-", 72)
        log.info("Access URLs:\n{}\n\tLocal:\t\thttp://localhost:{}{}\n{}", dashes, port, contextPath, dashes)
    }

}
